plenum/common/messages/fields.py

```diff
@@ -710,11 +711,11 @@ def _specific_validation(self, val):
         if len(val) != 4:
             return 'should have size of 4'
 
-        view_no, pp_view_no, pp_seq_no, pp_digest = val
-        for validator, value in ((NonNegativeNumberField().validate, view_no),
-                                 (NonNegativeNumberField().validate, pp_view_no),
-                                 (NonNegativeNumberField().validate, pp_seq_no),
-                                 (NonEmptyStringField().validate, pp_digest)):
+        bid = BatchID(*val)
```

Comment:
1. This line can throw, and as far as I understand caller of `_specific_validation` doesn't expect exceptions. If so it needs to be fixed to just return an error (either by wrapping in try/except or adding additional checks)
2. Serialized BatchID should be dictionary, not list.